prov/gni/src/gnix_msg.c

```diff
@@ -1044,6 +1050,8 @@ static int __smsg_rndzv_fin(void *data, void *msg)
 				"GNI_SmsgRelease returned %s\n",
 				gni_err_str[status]);
 		ret = gnixu_to_fi_errno(status);
+	} else {
+		ret = gnixu_to_fi_errno(hdr->status);
```

Comment:
Returning an error here causes us to stop processing messages on the SMSG connection.  I don't think that's what we want.

gnix_nic.c
1607                 if (status == GNI_RC_SUCCESS) {
1608                         GNIX_INFO(FI_LOG_EP_DATA, "Found RX (%p)\n", vc);
1609                         ret = nic->smsg_callbacks[tag](vc, msg_ptr);
1610                         if (ret != FI_SUCCESS) {
1611                                 /* Stalled, reschedule */
1612                                 break;
1613                         }